subsys/fs/nvs/Kconfig

```diff
@@ -29,6 +29,14 @@ config NVS_LOOKUP_CACHE_SIZE
 	  Number of entries in Non-volatile Storage lookup cache.
 	  It is recommended that it be a power of 2.
 
+config NVS_DATA_CRC
+	bool "Non-volatile Storage CRC protection on the data"
+	help
+	  Enable a CRC-32 on the data part of each NVS element.
+	  The ATE CRC is not impacted by this feature and stays the same.
+	  The CRC-32 is transparently stored at the end of the data field,
+	  in the NVS data section, so 4 more bytes are needed per NVS element.
```

Comment:
not 4 more bytes, but an extra write-block is needed

Reply:
The idea is to write the CRC just after the data, taking alignment into account. I'm closing this conversation because we found a way to achieve this.